src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -27,36 +33,66 @@ function LineVis(props: Props): ReactElement {
     curveType = CurveType.LineOnly,
     showGrid = true,
     scaleType = ScaleType.Linear,
+    abscissas,
   } = props;
 
-  const indexDomain = extendDomain([0, dataArray.size - 1], 0.01);
-  const dataDomain = useMemo(() => {
-    return domain
-      ? extendDomain(domain, 0.05, scaleType === ScaleType.Log)
+  const abscissaValues = abscissas || range(dataArray.size);
+  const abscissaToIndex = getValueToIndexScale(
+    abscissaValues.map(
+      (x, i) =>
+        abscissaValues[i - 1] + (abscissaValues[i] - abscissaValues[i - 1]) / 2
+    )
+  ); // Shifting index change threshold to midpoints
```

Comment:
Explained in greater details at https://github.com/silx-kit/h5web/pull/251#discussion_r506201340

The `map` operation gives `NaN` for `i=0` but this is irrelevant as the first value of the array is discarded when constructing the `ScaleThreshold`.

Reply:
Great solution, but seeing how it works for the Heatmap without any changes, I wonder... could we do something like this?

```js
const abscissaToIndex = getValueToIndexScale([...abscissaValues, NaN]);
```

Adding a fake value at the end of the array would shift all the thresholds like on the heatmap, wouldn't it? The last number would not appear on the axis since it's not part of `abscissaValues`, and it wouldn't appear in the tooltip since I think the last pixel of the plot would match the last real abscissa value.

Reply:
This won't work and I explain below.

The argument of `getValueToIndexScale` is directly the domain of the `scaleThreshold`. This domain contains the values of the threshold, period. There is no shift, the scale separates the `range` at the values given in `domain` (that can be arbitrary).

It works without tricks in the `Heatmap` as we want the tooltip there to switch between two values when crossing the edges that are the abscissa values.

In the `LineVis` however, we want to trigger the switch at the midpoint between two neighbouring abscissas. As `scaleThreshold` takes directly the values where the switch should appear, there is no way around creating the array of midpoints. 

To have the behaviour you suggest, we could use related scales `scaleQuantize` and `scaleQuantile`. I tried for some time but could not really get what I wanted (or it would involve circumvolutions). I settled in the end for `scaleThreshold` where I set the thresholds explicitly that I preferred over twisting around the other scales.

Reply:
Right, sorry I was picturing this working with `scaleQuantile`, but what you're saying makes sense. I've just played with the scales a bit to try to understand them a bit better and `scaleThreshold` does make the most sense.